### ISSUE TITLE
Missing adding a todo in a list item at the start of the line.

### ISSUE THREAD

**Issue by talw** [vitalk/vim-simple-todo#31]
I'm trying to add this to your plugin, this is where I'm at:

```viml
nnore <silent> <Plug>(simple-todo-new-list-item-start-of-line) mzI<c-r>=<SID>get_list_marker(line('.')-1)<cr><Esc>"=g:simple_todo_list_symbol.' [ ] '<cr>pa<space><Esc>`z4l
```

However, it adds the list item at the second position of the line instead of the first.
What is missing to get this to work?
I can make a pull request with the addition after it works if you're busy :)

Thanks!

**Comment by vitalk** [comment]
Hi, @talw!

I'm a little bit confused and don't clearly understand your issue, the plugin already has an API method to create a new item at start of the current line

```viml
<Plug>(simple-todo-new-start-of-line)
```

which is mapped to <kbd>,</kbd><kbd>I</kbd> by default.

**Comment by talw** [comment]
Hi, @vitalk!

I was trying to get it to put a **list item** with a check box, rather than just a checkbox, as that's what markdown works with. If I just put a checkbox by itself at the start of each line they would all appear in the same line in the resulting html.

This plugin can add a list item check box only in the cursor position and not at the start of the new line.
So essentially, I want to combine the `putting something at the start of the line` and the `putting a list item checkbox`

**Comment by vitalk** [comment]
Well,

what you trying to achieve is

```viml
nnore <silent> <Plug>(simple-todo-new-list-item-start-of-line) mzI<c-r>=g:simple_todo_list_symbol<cr><space>[ ]<space><Esc>`z6l
inore <silent> <Plug>(simple-todo-new-list-item-start-of-line) <Esc>mzI<c-r>=g:simple_todo_list_symbol<cr><space>[ ]<space><Esc>`z6la
```

The visual mode mapping require more time to investigate. I would be glad to accept a PR.

**Comment by talw** [comment]
Thanks for those lines!
I've tested them and they work!

Visual mode is a nice to have but not a must as you can repeat the command with dot perfectly.
Having said that, once I get around to learn more vimscript, this will be one of the first tasks I will attempt.

I would be happy to have this upstream, shall I do a PR with your change? Or do you already plan to insert it in one of your commits?

**Comment by vitalk** [comment]
@talw, PR is welcome (don't forgot to update documentation).

**Comment by vitalk** [comment]
Closed via #32
